src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceMidEventHandler.java

```diff
@@ -29,32 +42,82 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
     }
 
     @Override
-    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
         if (!canHandle(callbackType, callback)) {
             throw new IllegalStateException("Cannot handle callback");
         }
 
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         final SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
-        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse =
+            new PreSubmitCallbackResponse<>(sscsCaseData);
 
-        buildSscsDocumentFromScan(sscsCaseData, callback.isIgnoreWarnings(), preSubmitCallbackResponse);
+        buildSscsDocumentFromScan(sscsCaseData, caseDetails.getId(), callback.isIgnoreWarnings(),
+            preSubmitCallbackResponse);
 
         return preSubmitCallbackResponse;
     }
 
-    private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, Boolean ignoreWarnings, PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse) {
+    private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, long caseId, Boolean ignoreWarnings,
+                                           PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse) {
 
         if (isNotEmpty(sscsCaseData.getScannedDocuments())) {
+            List<String> unreadablePdfLinks = new ArrayList<>();
+            List<String> encryptedPdfLinks = new ArrayList<>();
             for (ScannedDocument scannedDocument : sscsCaseData.getScannedDocuments()) {
                 if (scannedDocument != null && scannedDocument.getValue() != null) {
-                    checkWarningsAndErrors(sscsCaseData, scannedDocument, sscsCaseData.getCcdCaseId(), ignoreWarnings, preSubmitCallbackResponse);
+                    if (scannedDocument.getValue().getUrl() != null
+                        && scannedDocument.getValue().getUrl().getDocumentUrl() != null) {
+                        try {
+                            footerService.isReadablePdf(scannedDocument.getValue().getUrl().getDocumentUrl());
+                        } catch (PdfPasswordException e) {
```

Comment:
This is API is not part of this PR, I know, but this is assuming its going to return a boolean you don't check, and it's throwing two different Exceptions for a password and unreadable pdf. You can't do anything with an API like this apart from write ugly code around it.
It would be better if it returned an enum for PDF_OK, UNREADABLE or HAS_PASSWORD.

Reply:
I agree but it is also used in UploadDocumentFurtherEvidenceAboutToSubmitHandler and they rely on catching exception too. If I change the API, we will have to test the other feature as well. The boolean return is not used by the methods.

Reply:
Refactored the code